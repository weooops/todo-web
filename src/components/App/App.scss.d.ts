export const app: string;
export const header: string;
export const blue: string;
