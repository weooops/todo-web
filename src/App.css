.App {
  text-align: center;
}

.Image__list {
  display: flex;
  flex-wrap: wrap;
}
.Image__item {
  width: 300px;
}
.Image__item img {
  width: 100%;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
