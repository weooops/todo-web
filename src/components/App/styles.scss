.app {
  margin: 0 auto;
  border: 1px solid green;
  text-align: center;
}

.container {
  max-width: 980px;
}