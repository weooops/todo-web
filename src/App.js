import Amplify, { I18n, Storage, API, graphqlOperation } from 'aws-amplify';
import { withAuthenticator, S3Image } from 'aws-amplify-react';
import gql from 'graphql-tag';
import { Query, Mutation } from 'react-apollo';
import React, { Component } from 'react';
import aws_exports from './aws-exports';
import { listTodos } from './graphql/queries';
import { createTodo } from './graphql/mutations';
import './App.css';

Amplify.configure(aws_exports);
Storage.configure({ level: 'private' });

const GET_TODOS = gql`
  ${listTodos}
`;

const ADD_TODO = gql`
  ${createTodo}
`;

class App extends Component {
  state = {
    name: '',
    description: '',
    todos: [],
    items: []
  };

  componentDidMount() {
    this.listQuery();
    Storage.list('', { level: 'private' })
      .then(result => this.setState({ items: result }))
      .catch(err => console.log(err));
  }

  render() {
    return (
      <div className="App">
        <h2>{I18n.get('Todos with Amplify Client')}</h2>
        <div>
          <button onClick={this.listQuery}>{I18n.get('GraphQL Query')}</button>
          <button onClick={this.todoMutation}>
            {I18n.get('GraphQL Mutation')}
          </button>
          <input
            type="text"
            name="name"
            placeholder={I18n.get('name')}
            value={this.state.name}
            onChange={this.handleChange}
          />
          <input
            type="text"
            name="description"
            placeholder={I18n.get('description')}
            value={this.state.description}
            onChange={this.handleChange}
          />
        </div>
        <div>
          {this.state.todos.map(todo => (
            <p key={todo.id}>
              <strong>{todo.name}</strong>, {todo.description},{' '}
              <small>{todo.createdAt}</small>
            </p>
          ))}
        </div>
        <hr />

        <h2>{I18n.get('Todos with Apollo Client')}</h2>
        <div>
          <Mutation
            mutation={ADD_TODO}
            variables={{ filter: { createdAt: 'createdAt' } }}
          >
            {(addTodo, { data }) => (
              <div>
                <form
                  onSubmit={e => {
                    e.preventDefault();
                    addTodo({
                      variables: {
                        input: {
                          name: this.state.name,
                          description: this.state.description
                        }
                      }
                    });
                  }}
                >
                  <input
                    type="text"
                    name="name"
                    placeholder={I18n.get('name')}
                    value={this.state.name}
                    onChange={this.handleChange}
                  />
                  <input
                    type="text"
                    name="description"
                    placeholder={I18n.get('description')}
                    value={this.state.description}
                    onChange={this.handleChange}
                  />
                  <button type="submit">Add Todo</button>
                </form>
              </div>
            )}
          </Mutation>
        </div>
        <div>
          <Query query={GET_TODOS}>
            {({ loading, error, data }) => {
              if (loading) return `${I18n.get('Loading')}...`;
              if (error) return `${I18n.get('Error')}! ${error.message}`;

              return (
                <div>
                  {data.listTodos.items.map(todo => (
                    <p key={todo.id}>
                      <strong>{todo.name}</strong>, {todo.description},{' '}
                      <small>{todo.createdAt}</small>
                    </p>
                  ))}
                </div>
              );
            }}
          </Query>
        </div>
        <hr />

        <h2>{I18n.get('Pick a file')}</h2>
        <div>
          <input type="file" onChange={this.uploadFile} />
          <div className="Image__list">
            {this.state.items.map(item => (
              <div className="Image__item" key={item.key}>
                <S3Image level="private" imgKey={item.key} />
              </div>
            ))}
          </div>
        </div>
      </div>
    );
  }

  uploadFile = e => {
    const file = e.target.files[0];
    const name = file.name;

    Storage.put(name, file).then(item => {
      this.setState({ items: [...this.state.items, item] });
    });
  };

  handleChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  todoMutation = async () => {
    const todoDetails = {
      input: {
        name: this.state.name,
        description: this.state.description
      }
    };

    const newTodo = await API.graphql(
      graphqlOperation(createTodo, todoDetails)
    );
    this.setState({
      name: '',
      description: '',
      todos: [newTodo.data.createTodo, ...this.state.todos]
    });
  };

  listQuery = async () => {
    const allTodos = await API.graphql(
      graphqlOperation(listTodos, { variables: { filter: 'name' } })
    );
    this.setState({ todos: allTodos.data.listTodos.items });
  };
}

export default withAuthenticator(App, true);
