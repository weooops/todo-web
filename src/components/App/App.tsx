import * as React from 'react';

import * as styles from './App.scss';
import UserList from '../UserList';

export interface UserType {
  id: number;
  username: string;
}

interface StatesType {
  input: string;
  users: Array<UserType>;
}

class App extends React.Component<{}, StatesType> {
  id = 3;
  state = {
    input: '',
    users: [
      { id: 1, username: 'jake' },
      { id: 2, username: 'jason' }
    ]
  };

  onInputChange = (e: React.FormEvent<HTMLInputElement>): void => {
    const target = e.target as HTMLInputElement;
    this.setState({ input: target.value });
  }

  onButtonClick = () => {
    this.setState(({ input, users }) => ({
      input: '',
      users: users.concat({
        id: this.id++,
        username: input
      })
    }));  
  }

  render() {
    console.log('App이 렌더링되고 있어요.');

    return (
      <div className={styles.app}>
        <div className={styles.header}>
          <input onChange={this.onInputChange} value={this.state.input} />
          <button onClick={this.onButtonClick}>추가</button>
        </div>
        <h1 className={styles.blue}>사용자 목록</h1>
        <div>
          <UserList users={this.state.users} />
        </div>
      </div>
    );
  }
}

export default App;