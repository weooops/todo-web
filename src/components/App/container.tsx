import * as React from 'react';

import { UserType, TodoType } from '../../models/';
import App from './App';

interface PropsType {
  users: Array<UserType>;
  todos: Array<TodoType>;
  getUsers(): void;
  getTodos(): void;
}

interface StateType {
  loading: boolean;
  input: string;
}

class Container extends React.Component<PropsType, StateType> {
  state: StateType = {
    loading: true,
    input: 'mark'
  };

  componentDidMount(): void {
    const { users, getUsers, todos, getTodos } = this.props;

    if (users.length && todos.length) {
      this.setState({ loading: false });
    } else {
      getUsers();
      getTodos();
    }
  }

  componentWillReceiveProps(nextProps: PropsType): void {
    if (nextProps.users.length && nextProps.todos.length) {
      this.setState({ loading: false });
    }
  }

  render(): JSX.Element {
    const { users, todos } = this.props;
    const { loading, input } = this.state;

    return (
      <App
        users={users}
        loading={loading}
        input={input}
        onInputChange={this._onInputChange}
        todos={todos}
      />
    );
  }

  private _onInputChange = (e: React.FormEvent<HTMLInputElement>): void => {
    const { value } = e.currentTarget;
    this.setState({ input: value });
  }
}

export default Container;
