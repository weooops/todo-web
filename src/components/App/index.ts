import App, { UserType } from './App';

export default App;
export { UserType };