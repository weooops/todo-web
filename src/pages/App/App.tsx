import * as React from 'react';

import * as styles from './styles.scss';
import * as classNames from 'classnames/bind';
import Navigation from '../../containers/Navigation';
import Private from '../../containers/Private';
import Public from '../../containers/Public';

const cx = classNames.bind(styles);

interface IProps {
  isLoggedIn: boolean;
}

const App: React.SFC<IProps> = props => (
  <div className={cx('app', 'container')}>
    {props.isLoggedIn && <Navigation key={3} />}
    {props.isLoggedIn ? <Private key={1} /> : <Public key={2} />}
  </div>
);

export default App;
