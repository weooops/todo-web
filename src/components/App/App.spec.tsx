import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import App from './';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

it('renders correctly', () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});