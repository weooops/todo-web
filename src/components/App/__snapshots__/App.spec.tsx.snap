// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className={undefined}
        >
          <input
            onChange={[Function]}
            value=""
          />
          <button
            onClick={[Function]}
          >
            추가
          </button>
        </div>,
        <h1
          className={undefined}
        >
          사용자 목록
        </h1>,
        <div>
          <UserList
            users={
              Array [
                Object {
                  "id": 1,
                  "username": "jake",
                },
                Object {
                  "id": 2,
                  "username": "jason",
                },
              ]
            }
          />
        </div>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              onChange={[Function]}
              value=""
            />,
            <button
              onClick={[Function]}
            >
              추가
            </button>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "추가",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "추가",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "사용자 목록",
          "className": undefined,
        },
        "ref": null,
        "rendered": "사용자 목록",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <UserList
            users={
              Array [
                Object {
                  "id": 1,
                  "username": "jake",
                },
                Object {
                  "id": 2,
                  "username": "jason",
                },
              ]
            }
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "users": Array [
              Object {
                "id": 1,
                "username": "jake",
              },
              Object {
                "id": 2,
                "username": "jason",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className={undefined}
          >
            <input
              onChange={[Function]}
              value=""
            />
            <button
              onClick={[Function]}
            >
              추가
            </button>
          </div>,
          <h1
            className={undefined}
          >
            사용자 목록
          </h1>,
          <div>
            <UserList
              users={
                Array [
                  Object {
                    "id": 1,
                    "username": "jake",
                  },
                  Object {
                    "id": 2,
                    "username": "jason",
                  },
                ]
              }
            />
          </div>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                onChange={[Function]}
                value=""
              />,
              <button
                onClick={[Function]}
              >
                추가
              </button>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "추가",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "추가",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "사용자 목록",
            "className": undefined,
          },
          "ref": null,
          "rendered": "사용자 목록",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <UserList
              users={
                Array [
                  Object {
                    "id": 1,
                    "username": "jake",
                  },
                  Object {
                    "id": 2,
                    "username": "jason",
                  },
                ]
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "users": Array [
                Object {
                  "id": 1,
                  "username": "jake",
                },
                Object {
                  "id": 2,
                  "username": "jason",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
